### ISSUE TITLE
Improve handling of changed behavior in the game

### ISSUE THREAD

**Issue by maxteufel** [SuperTux/supertux#474]
In order to prevent broken levels caused due to a change of behavior in gameplay, a new option could be introduced for levels to allow the author to specify features used/needed.

Example:
```lisp
(level-features
  (swimming #t)
  (glow-effects #f)
)
```

**Comment by Karkus476** [comment]
@maxteufel Initially I liked this idea, but I am concerned that it will introduce too much inconsistency between behavior of different levels.

**Comment by maxteufel** [comment]
@Karkus476 That's true. Could we inform/warn/yell at the user if the settings for a certain level aren't consistent with 'official settings'? Because I also don't think a hard transition from no swimming to swimming is the way to go.

**Comment by Hume2** [comment]
I have an idea: All levels saved by the in-game editor could use only the actual settings, so it wouldn't be possible to change the settings for most of the players. If there was ever made a server for accepting add-ons, it could deny all levels those don't use the most actual settings.

**Comment by Karkus476** [comment]
@maxteufel @Hume2 personally I think it may cause confusion, and most levels will need redesigning anyway, as the game progresses

**Comment by Ordoviz** [comment]
#1193 suggests saving a version stamp in level files.

**Comment by Alasdairbugs** [comment]
Closed as dev notes have been added
